Test/git-issues/git-issue-2593.dfy.expect

```diff
@@ -0,0 +1,4 @@
+git-issue-2593.dfy(7,24): Warning: Argument to 'old' does not dereference the mutable heap, so this use of 'old' has no effect
+git-issue-2593.dfy(7,10): Warning: /!\ No terms found to trigger on.
```

Comment:
Shouldn't we produce `P(y)` as the trigger here?

Reply:
But in this case old is redundant, right?  I guess maybe it's not an issue.
But is `old()` shouldn't always be a trigger killer, right?  Like `old(c.x[i])` is a reasonable trigger for `i`.

Reply:
> But in this case old is redundant, right? I guess maybe it's not an issue.

I don't think it's an issue either. If there was a warning, then any expression enclosing old() would not be considered as a trigger, but after fixing this warning, all the other triggers you dreamed of become available.

> But is old() shouldn't always be a trigger killer, right? Like old(c.x[i]) is a reasonable trigger for i.

Right. `old()` become a killer trigger only when it's marked as useless, which happens inside the rewriter check. So your example would work as usual, since it would not generate any warning.

Reply:
But why do we bother marking it as a trigger killer when it's useless?

Reply:
Let me rephrase. This PR's fix is to prevent old(X) becoming a trigger if old(X) was previously marked as useless. The enclosing function is `IsPotentialTriggerCandidate` and will just return `false` if that is the case (previously, it returned true). That's the only change that this PR does.
https://github.com/dafny-lang/dafny/blob/4d65937ebc88c7371cb598c4005cddb82b4ff140/Source/DafnyCore/Triggers/TriggersCollector.cs#L243-L254

Now, it just happens that the calling function has a few exceptions in the case `IsPotentialTriggerCandidate` returns false, to not mark something as a trigger killer (the second argument of `AnnotateOther`)
https://github.com/dafny-lang/dafny/blob/4d65937ebc88c7371cb598c4005cddb82b4ff140/Source/DafnyCore/Triggers/TriggersCollector.cs#L212-L234
I just did not bother adding `old()` as an exception. Why would we bother adding it as an exception when it's already a warning?